cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java

```diff
@@ -376,7 +377,8 @@ public void apply(MetadataDataset input) throws Exception {
   public MetadataSearchResponse search(String namespaceId, String searchQuery,
                                        Set<EntityTypeSimpleName> types,
                                        SortInfo sortInfo, int offset, int limit,
-                                       int numCursors, String cursor, boolean showHidden) throws BadRequestException {
+                                       int numCursors, String cursor, boolean showHidden,
+                                       Set<EntityScope> entityScope) throws BadRequestException {
```

Comment:
Initially, I had it as `EnumSet<EntityScope>`, but I searched some use of enumset, and found that some places are passing Set instead of EnumSet as parameter types, for example, https://github.com/caskdata/cdap/blob/release/4.1/cdap-security-spi/src/main/java/co/cask/cdap/security/spi/authorization/PrivilegesManager.java#L46. I feel using a Set provides a more general way to call that method, like passing`ImmutableSet.of(EntityScope.USER, EntityScope.SYSTEM)`